sql-plugin/src/main/scala/org/apache/spark/sql/rapids/AggregateFunctions.scala

```diff
@@ -891,3 +932,118 @@ case class GpuCollectSet(
       inputs: Seq[(ColumnVector, Int)]): RollingAggregationOnColumn =
     RollingAggregation.collectSet().onColumn(inputs.head._2)
 }
+
+/**
+ * TODO
+ */
+abstract class GpuM2(child: Expression)
+  extends GpuAggregateFunction with ImplicitCastInputTypes {
+
+  // TODO: remove comment below
+  // dataType must be a structs column, and nullable of the output must be false
+  // the struct field `mean` and `m2` can be nullable (but are not always nullable)
+  // so I'm no sure if set `nullable = true` is correct here.
+
+  override def dataType: DataType = StructType(Array(
```

Comment:
nit: this struct type is defined in two places and it would be good to put it in a constant